### PR TITLE
feat(blocks): add plain text adapter for database block

### PR DESCRIPTION
**Pull request by donteatfriedrice**
[BS-2059](https://linear.app/affine-design/issue/BS-2059/补齐-database-plain-text-adapter)

### CODE REVIEW

**Comment by donteatfriedrice** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/9039?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9039** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/9039?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/9039?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9036** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/9036?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9039).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/blocks/src/database-block/adapters/plain-text.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                default:
+                  return String(cell.value);
+              }
+            })
+        );
+
+        // Handle first row.
+        if (Array.isArray(columns)) {
+          rows.push(
+            Array.prototype.map.call(columns, v => String(v.name)) as string[]
```
Comment (fundon) [inline packages/blocks/src/database-block/adapters/plain-text.ts:79]: use `columns.map` directly

File: packages/blocks/src/database-block/adapters/plain-text.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        const { walkerContext, deltaConverter } = context;
+        const rows: string[][] = [];
+        const columns = o.node.props.columns as Array<Column>;
+        const children = o.node.children;
+        const cells = o.node.props.cells as SerializedCells;
+        const tableCells = Array.prototype.map.call(
+          children,
+          (v: BlockSnapshot) =>
+            Array.prototype.map.call(columns, col => {
+              const cell = cells[v.id]?.[col.id];
```
Comment (fundon) [inline packages/blocks/src/database-block/adapters/plain-text.ts:34]: use `columns.map` directly

File: packages/blocks/src/database-block/adapters/plain-text.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    fromMatch: o => o.node.flavour === DatabaseBlockSchema.model.flavour,
+    toBlockSnapshot: {},
+    fromBlockSnapshot: {
+      enter: (o, context) => {
+        const { walkerContext, deltaConverter } = context;
+        const rows: string[][] = [];
+        const columns = o.node.props.columns as Array<Column>;
+        const children = o.node.children;
+        const cells = o.node.props.cells as SerializedCells;
+        const tableCells = Array.prototype.map.call(
```
Comment (fundon) [inline packages/blocks/src/database-block/adapters/plain-text.ts:30]: use `columns.map` directly

File: packages/blocks/src/database-block/adapters/plain-text.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          children,
+          (v: BlockSnapshot) =>
+            Array.prototype.map.call(columns, col => {
```
Comment (fundon) [inline packages/blocks/src/database-block/adapters/plain-text.ts:33]: use `columns.map` directly

File: packages/blocks/src/database-block/adapters/plain-text.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                case 'date':
+                  return format(new Date(cell.value as number), 'yyyy-MM-dd');
+                case 'select': {
+                  const value = col.data.options.find(
+                    (opt: Record<string, string>) => opt.id === cell.value
+                  )?.value;
+                  return value || '';
+                }
+                case 'multi-select': {
+                  const value = Array.prototype.map
```
Comment (fundon) [inline packages/blocks/src/database-block/adapters/plain-text.ts:58]: same here

File: packages/blocks/src/database-block/adapters/plain-text.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+              case 'date':
+                return format(new Date(cell.value as number), 'yyyy-MM-dd');
+              case 'select': {
+                const value = (
+                  col.data as { options: Array<Record<string, string>> }
+                ).options.find(opt => opt.id === cell.value)?.value;
+                return value || '';
+              }
+              case 'multi-select': {
+                const value = Array.prototype.map
```
Comment (fundon) [inline packages/blocks/src/database-block/adapters/plain-text.ts:54]: use map directly here

**Review by fundon — APPROVED** [review]

**Pull request closed without merging**